handlers/recipes.go

```diff
@@ -0,0 +1,93 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/Daniorocket/cookit-backend/models"
+	uuid "github.com/satori/go.uuid"
+	"gopkg.in/validator.v2"
+)
+
+func (d *Handler) CreateRecipe(w http.ResponseWriter, r *http.Request) {
+	tkn := r.Context().Value("token").(JWT)
+	recipe := models.Recipe{}
+	recipe.UserID = tkn.Username
+	recipe.ID = uuid.NewV4().String()
+	recipe.Date = time.Now().UTC().String()
+	fmt.Println("username:", tkn.Username, "recipe:", recipe)
+	if err := json.NewDecoder(r.Body).Decode(&recipe); err != nil {
+		// If there is something wrong with the request body, return a 400 status
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to decode json"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	if err := validator.Validate(recipe); err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to validate json"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	if err := models.CreateRecipe(d.Client, d.DatabaseName, &recipe); err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to create recipe"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	CreateApiResponse(w, nil, http.StatusOK, "success", "none")
+}
+func (d *Handler) GetListOfRecipes(w http.ResponseWriter, r *http.Request) {
+	page := r.URL.Query().Get("page")
+	limit := r.URL.Query().Get("limit")
+	recipes, te, err := models.GetAllRecipes(d.Client, d.DatabaseName, page, limit)
+	if err != nil {
+		if err := CreateApiResponse(w, nil, http.StatusInternalServerError, "failed", "Failed to get list of recipes"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		log.Println("Error:", err)
+		return
+	}
+	if err := CreateApiResponse(w, Pagination{
+		Data:          recipes,
+		Limit:         limit,
+		Page:          page,
+		TotalElements: te,
+	}, http.StatusOK, "success", "none"); err != nil {
+		log.Println("Error create Api response:", err)
+	}
+}
+func (d *Handler) GetListOfRecipesByTags(w http.ResponseWriter, r *http.Request) {
+	data := make(map[string][]int)
```

Comment:
Czemu do mapy, skoro później jednego klucza tylko używasz?

Reply:
Gdy nie dekodowałem tagów do mapy to program mi tu rzucał error decode'a. Tak wyglądał JSON dla tego handlera.
 {
    "tags": [1,2]
 }

Reply:
https://mholt.github.io/json-to-go/
```go
type AutoGenerated struct {
	Tags []int `json:"tags"`
}
```

Reply:
Aa, już rozumiem, myślałem, że się nie tworzy struktury, kiedy jest tylko jedno pole do zdekodowania i da się to zrobić mapą. Robiłem tak w innych przypadkach, dziękuję poprawiam.

Reply:
Przewaga struktury nad mapą w tym przypadku jest to, że od razu wiesz z czego się składa request, a tak żeby dojść że jest tam tylko `tags` musiałem wgłębić się w metodę